pkg/controller/releasemanager/syncer.go

```diff
@@ -206,6 +209,30 @@ func (r *ResourceSyncer) syncApp(ctx context.Context) error {
 	return nil
 }
 
+func (r *ResourceSyncer) syncCanary(ctx context.Context) error {
```

Comment:
why not call this in the state machine? then you don't need the `incarnationsInStates` dance.

Reply:
Ah got it, thanks, will do